crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -922,6 +924,87 @@ end
 # DATA LOADERS
 # =================================================================================================
 
+#! Loads account data from the advice inputs into the _current_ account's memory section.
+#!
+#! Inputs:
+#!   Operand stack: [account_id_prefix, account_id_suffix]
+#!   Advice map: {
+#!     ACCOUNT_ID: [[account_id_suffix, account_id_prefix, 0, account_nonce],
+#!                          VAULT_ROOT, STORAGE_COMMITMENT, CODE_COMMITMENT],
+#!     STORAGE_COMMITMENT: [[STORAGE_SLOT_DATA]],
+#!     CODE_COMMITMENT: [num_procs, [ACCOUNT_PROCEDURE_DATA]]
+#!   }
+#! Outputs:
+#!   Operand stack: []
+#!
+#! Where:
+#! - account_id_{prefix,suffix} are the prefix and suffix felts of the ID of the account.
+#! - ACCOUNT_ID is the word constructed from the account_id as follows:
+#!   [account_id_suffix, account_id_prefix, 0, 0].
+#! - account_nonce is the nonce of the account.
+#! - VAULT_ROOT is the commitment of the account's vault.
+#! - STORAGE_COMMITMENT is the commitment to the account's storage.
+#! - STORAGE_SLOT_DATA is the data contained in the storage slot which is constructed as follows:
+#!   [SLOT_VALUE, slot_type, 0, 0, 0].
+#! - CODE_COMMITMENT is the commitment to the account's code.
+#! - ACCOUNT_PROCEDURE_DATA is the information about account procedures which is constructed as
+#!   follows: [PROCEDURE_MAST_ROOT, storage_offset, 0, 0, 0].
+#!
+#! Panics if:
+#! - the number of account procedures exceeded the maximum limit of 256.
+#! - the computed account code commitment does not match the provided account code commitment.
+#! - the number of account storage slots exceeded the maximum limit of 255.
+#! - the computed account storage commitment does not match the provided account storage commitment.
+export.load_foreign_account
+    emit.ACCOUNT_BEFORE_LOAD_FOREIGN
+    # => [account_id_prefix, account_id_suffix]
+
+    # construct the word with account ID to load the core account data from the advice map
+    push.0.0
+    # OS => [0, 0, account_id_prefix, account_id_suffix]
+
+    # move the core account data to the advice stack
+    adv.push_mapval
+    # OS => [0, 0, account_id_prefix, account_id_suffix]
+    # AS => [[account_id_prefix, account_id_suffix, 0, account_nonce], VAULT_ROOT, STORAGE_COMMITMENT, CODE_COMMITMENT]
+
+    # store the id and nonce of the foreign account to the memory
+    dropw adv_loadw
```

Comment:
Why do we need the `dropw` here? If I'm understanding correctly, the state of the stack before this `dropw` is:
```
[0, 0, account_id_prefix, account_id_suffix]
```
If so, then we could just do `adv_loadw` to overwrite this with the first word from the advice map.